functions/transferChatStart.ts

```diff
@@ -215,38 +215,117 @@ export const handler = TokenValidator(
         return;
       }
 
+      /**
+       * Conversations comes with attributes.conversations filled in.
+       * The code below can be simplified after all channels are moved to conversations.
+       */
+      const originalConversations = originalAttributes.conversations;
+      let conversations;
+
+      if (originalConversations) {
+        conversations = originalConversations;
+      } else if (originalAttributes.conversation) {
+        conversations = originalAttributes.conversation;
+      } else {
+        conversations = { conversation_id: taskSid };
+      }
+
       const newAttributes = {
         ...originalAttributes,
-        conversations: originalAttributes.conversation // set up attributes of the new task to link them to the original task in Flex Insights
-          ? originalAttributes.conversation
-          : { conversation_id: taskSid },
+        conversations, // set up attributes of the new task to link them to the original task in Flex Insights
         ignoreAgent, // update task attributes to ignore the agent who transferred the task
         targetSid, // update task attributes to include the required targetSid on the task (workerSid or a queueSid)
         transferTargetType,
       };
 
-      // Edit channel attributes so that original task won't cause issues with the transferred one
-      await setDummyChannel(context, {
```

Comment:
@murilovmachado -- you know about @GPaoloni's recent changes to fix transfers for queues, right? I know he changed some of this code to make it work. Just wanted to make sure you were considering this, and that we'll be doing regression testing to make sure it continues to work.

Reply:
@nickhurlburt  I'm not very aware, but this piece specifically that you highlighted came from the merge I Just did with master. So it must be exactly the change Gian's introduced.

Reply:
Okay. Here's the PR: https://github.com/techmatters/serverless/pull/622. All I'm asking is to be careful about introducing any regressions, since they can sneak in if we're not actively looking.